rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/BookKeeper.java

```diff
@@ -136,17 +136,17 @@ public BookKeeper(Configuration conf)
       for (long blockNum = startBlock; blockNum < endBlock; blockNum++) {
         totalRequests++;
         long split = (blockNum * blockSize) / splitSize;
-        if (md.isBlockCached(blockNum)) {
-          blockLocations.add(new BlockLocation(Location.CACHED, blockSplits.get(split)));
-          cachedRequests++;
+        if (!blockSplits.get(split).equalsIgnoreCase(nodeName)) {
```

Comment:
Is this just refactor or is there a functional impact of re-arranging this?

Reply:
Nope this is not just refactor. In the previous case when we were checking for the cached one (now we are checking for non-local one), md. isBlockCached method will go ahead and create the md file if it doesn't exist. So in this case we were also creating md files for files which won't be residing on that node. 

By checking non-local first we are making sure if the file is non going to reside on this node, we are not calling md.isBlockCached hence not creating the file.